core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -0,0 +1,179 @@
+package cats
+package data
+
+import scala.annotation.tailrec
+import scala.collection.immutable.VectorBuilder
+import cats.std.vector._
+
+/**
+ * A data type which represents a single element (head) and a Vector
+ * (tail).  This can be used to represent a Vector which is guaranteed
+ * to not be empty.
+ */
+final case class NonEmptyVector[A] private (vector: Vector[A]) {
+
+  def unwrap: Vector[A] = vector
+
+  def head: A = vector.head
+
+  def tail: Vector[A] = vector.tail
+
+  /**
+   * remove elements not matching the predicate
+   */
+  def filter(f: A => Boolean): Vector[A] = vector.filter(f)
+
+  /**
+   * Append another NonEmptyVector to this
+   */
+  def combine(other: NonEmptyVector[A]): NonEmptyVector[A] = NonEmptyVector(vector ++ other.vector)
```

Comment:
We probably also want a variant of this that works with `Vector[A]`, since appending a vector to a non-empty vector (or vice-versa) always produces a non-empty vector.